src/d3a/models/area/__init__.py

```diff
@@ -356,6 +356,10 @@ def cycle_markets(self, _trigger_event=True, _market_cycle=False, deactivate=Fal
         else:
             changed_balancing_market = None
 
+        if d3a.constants.CONNECT_TO_PROFILES_DB and not self.parent and self.next_market:
+            global_objects.profile_db_connection.buffer_profiles_from_db(
```

Comment:
global_objects is also accessible from the simulation.py IIRC, therefore you can move this there and 1. not have to check the if block for every area, and remove the not self.parent clause 2. move the call of the method somewhere where it is more appropriate (because the Area class conceptually does not have any relation to profile management).
In addition to that, it would be much better to expose a method in the global_objects, called update_on_market_cycle, that would encapsulate the d3a.constants.CONNECT_TO_PROFILES_DB check. That way the check will be in only one place.

Reply:
@spyrostz I remember now why this has to be exactly here (for this one I actually spent the most time): We need to rotate the profiles after the next_market is created, but before the market_cycle event is dispatched to all subareas. This is because of a hen-egg problem: If I dont check if the profile contains the `self.next_market.time_slot` before actually accessing it, an error will occur. But I can not rotate the file without knowing the `self.next_market.time_slot`. 
(I wanted to drop a in-code comment to show this, but forgot)
Do you have an idea how to fix this?
Just FYI: the profiles are only read once, because there is a check for "did i already read new profiles" in `buffer_profiles_from_db`. So even if you dont have an idea, it is ok to stay here. We only have to bite the bullet of the two unneeded if statements for each area. 
Looking forward to your comment.

Reply:
(ah, I remember now why I added the check for self.parent: In order to only run it for the root are and consequently only once per market slot)

Reply:
Understandable, but you really do not need the market in order to calculate the timeslot. You can easily circumvent that by creating a small method in the area class that exposes the now_value (badly named, but in reality it is the current datetime of the market that will be created, irregardless if we have a simulation or CN). Therefore all you have to do is to create a method that exposes the now_value (and name it appropriately, eg current_slot_datetime or current_timeslot) and once you have this method you can use it from the Simulation class, since the Simulation class has access to the root area (which is all you need since this is the only area without a parent)

Reply:
Ahh, thanks a lot! will do that.